src/NumericalAlgorithms/Spectral/Mesh.cpp

```diff
@@ -73,6 +73,17 @@ void Mesh<Dim>::pup(PUP::er& p) {
   p | quadratures_;
 }
 
+template <size_t Dim>
+bool is_isotropic(const Mesh<Dim>& mesh) {
```

Comment:
I know you like free functions but this looks like it should be a member function :) mostly for discoverability.

Reply:
(replying on this thread rather than in a new PR comment)

I understand some of the points made in the talk, but not all of them. I think `std::begin()` and `std::end()` are bad examples chosen in the talk because I find those very cumbersome to work with. And I don't understand the counter-argument to "Programmers cannot find free functions as easily as member functions". I think that's actually quite true and is my main concern with the "free all functions" paradigm. I'd know where look for `mesh.is_isotropic()` but I would not know where to find `is_isotropic(mesh)`.

To be clear, I'm not blocking anything here. I would have implemented `is_isotropic` as a member function, but I can live with the free function too :)

Reply:
I actually like `std::begin()`, I don't always use it, but that's because member versions are necessary for data encapsulation. The point is not to free _all_ functions, it is to free any function that doesn't directly need access to private state not otherwise accessible. This actually tightens the encapsulation. In terms of discoverability, this is actually a shortcoming of documentation, not of where to put the function. One solution would be to have a section in the class dox that lists associated free functions, then when someone adds a free function it gets listed. Remove one, it gets removed. I think it's important to not have shortcomings in documentation and documentation tooling (let's face it, Doxygen is okay but could be better) dictate how we write our code. There are major technical advantages to free functions over member functions, and so a solution should be found for documenting them, not choosing an inferior technical solution to document them.

I know :) I think it's a good discussion to have, though!